optuna/search_space/intersection.py

```diff
@@ -1,13 +1,55 @@
+from __future__ import annotations
+
 from collections import OrderedDict
 import copy
 from typing import Dict
-from typing import Optional
+from typing import List
+from typing import Tuple
 
 import optuna
 from optuna.distributions import BaseDistribution
 from optuna.study import Study
 
 
+def _calculate(
+    trials: List[optuna.trial.FrozenTrial],
+    include_pruned: bool = False,
+    search_space: Dict[str, BaseDistribution] | None = None,
+    cursor: int = -1,
```

Comment:
`cursor` is not very descriptive in the context of optuna. Not necessarily addressed in this PR.
```suggestion
    last_seen_(largest_)trial_number: int = -1,
```

Reply:
Good catch! I changed the variable name `cursor` to `cached_trial_number` and `next_cursor` to `next_cached_trial_number` because we use also the variable name `next_cursor` and `next_last_seen_trial_number` is a bit confusing name. What do you think?

Reply:
Looks good, thank you very much.